backend/services/llm/completion_service.py

```
@@ -282,26 +370,38 @@ async def execute_non_streaming_completion(
+                    if hasattr(choice.message, 'tool_calls') and choice.message.tool_calls:
+                        tool_calls_list = []
+                        for tc in choice.message.tool_calls:
+                            if hasattr(tc, 'model_dump'):
+                                tool_calls_list.append(tc.model_dump())  # type: ignore
+                            else:
+                                tool_calls_list.append(dict(tc))  # type: ignore
                 
-                if not content:
-                    self.logger.error(f"No content extracted from LiteLLM response")
+                if not content and not tool_calls_list:
+                    self.logger.error(f"No content or tool_calls extracted from LiteLLM response")
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)